### PR TITLE
fix #22, refactor getContract.js and add limit parameter for fat_db

### LINKED ISSUE #22: Empty output json file when using cmd get-state with --output parameter
The flag `--output` is not working and needs to be debugged.

### PR DESCRIPTION
**Pull request by thecodingshrimp**
PR for issue #22

### CODE REVIEW

**Review by informartin — CHANGES_REQUESTED** [review]
Looking good so far. 

Could you please also store the state of referenced contracts, as mentioned by you in issue #22?

Could you also add a test to the test suite, please? Thanks!

**Comment by thecodingshrimp** [comment]
Ok.

Then, I would like to change the json structure so that it is human readable and also easier to import for further processing.

`{ contract_address, raw_contract_code, state: { key: value }, state_references: [], static_references: [] }`

Where state_references and static_references are arrays of objects representing referenced contracts with the same structure as above.

**Comment by thecodingshrimp** [comment]
I had to stringify the arrays state_references and static_references because the internal JSON library wouldn't parse it otherwise. It thinks that there is a circle reference because referenced contracts have the same properties as the root object.

File: tests/test/testGetState.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const { execSync } = require('child_process');
+const SimpleSmartContractStorage = artifacts.require("SimpleSmartContractStorage");
+const ContractWithLib = artifacts.require("ContractWithLib");
+const Web3 = require('web3');
+const fs = require('fs');
+
+const source_dsl = 'http://localhost:8545';
+const jsonFileName = '../test.json';
```
Comment (informartin) [inline tests/test/testGetState.js]: When calling the test from the project directory, the test.json is stored one level higher and cannot be accessed afterwards:

```
1) Contract: testGetState
       should return all values/references in state and code to other contracts:
     Error: ENOENT: no such file or directory, open '../test.json'
```

**Review by informartin — CHANGES_REQUESTED** [review]

File: tests/test/testGetState.js (hunk 1 above)
Comment (thecodingshrimp) [reply tests/test/testGetState.js]: commited a fix

**Review by informartin — CHANGES_REQUESTED** [review]
There seems to be an issue in the assertion in the test. When I execute the test, I receive:

```
 1 failing

  1) Contract: testGetState
       should return all values/references in state and code to other contracts:

      AssertionError: expected { Object (contract_address, raw_contract_code, ...) } to deeply equal { Object (contract_address, raw_contract_code, ...) }
      + expected - actual

       {
         "contract_address": "0x8235d814236b0A94F7c7b31C25D55F31eFB9B8d1"
      -  "raw_contract_code": "0x608060405234801561001057600080fd5b50600436106100365760003560e01c806375f890ab1461003b578063958f85bd1461007f575b600080fd5b61007d6004803603602081101561005157600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff1690602001909291905050506100c9565b005b61008761010d565b604051808273ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b80600160006101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555050565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1690509056fea2646970667358221220f80c17cb9e7fcb0d5aba153c31a1fb5ac4151a2ac3d17505516e3f3886fc0a1564736f6c63430006000033"
      +  "raw_contract_code": "0x608060405234801561001057600080fd5b50600436106100365760003560e01c806375f890ab1461003b578063958f85bd1461007f575b600080fd5b61007d6004803603602081101561005157600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff1690602001909291905050506100c9565b005b61008761010d565b604051808273ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200191505060405180910390f35b80600160006101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff16021790555050565b6000600160009054906101000a900473ffffffffffffffffffffffffffffffffffffffff1690509056fea2646970667358221220247882a5fcc13732397157e4c3f7102305f02261ed2e56e5144b569bf681997264736f6c63430006000033"
         "state": {
           "0000000000000000000000000000000000000000000000000000000000000000": "000000000000000000000000009a77b77c1bd10f6bd6ce0c76b6a06c1df5e8ae"
         }
         "state_references": [
      -    "{\n    \"contract_address\": \"0x663144906fD032B4A3F8c8726c8Eb36EC9c860B5\",\n    \"raw_contract_code\": \"0x608060405234801561001057600080fd5b50600436106100415760003560e01c80637bd703e81461004657806390b98a111461009e578063f8b2cb4f14610104575b600080fd5b6100886004803603602081101561005c57600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff16906020019092919050505061015c565b6040518082815260200191505060405180910390f35b6100ea600480360360408110156100b457600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff16906020019092919080359060200190929190505050610200565b604051808215151515815260200191505060405180910390f35b6101466004803603602081101561011a57600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff169060200190929190505050610359565b6040518082815260200191505060405180910390f35b600073bf9f4b85ccec7d129257868c59d8968f210bcd7e6396e4ee3d61018184610359565b60026040518363ffffffff1660e01b8152600401808381526020018281526020019250505060206040518083038186803b1580156101be57600080fd5b505af41580156101d2573d6000803e3d6000fd5b505050506040513d60208110156101e857600080fd5b81019080805190602001909291905050509050919050565b6000816000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000205410156102515760009050610353565b816000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008282540392505081905550816000808573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020600082825401925050819055508273ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff167fddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef846040518082815260200191505060405180910390a3600190505b92915050565b60008060008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002054905091905056fea2646970667358221220ab4254d8168dfaa1ecab664abd46967753721498b42b5636dac3f77f1017e8e564736f6c63430006000033\",\n    \"state\": {\n        \"c02b974894de5bf0fb7566ce8f7b1adad2756368026c4d4ee0415f5e0642cc68\": \"0000000000000000000000000000000000000000000000000000000000002710\"\n    },\n    \"static_references\": [\n        \"{\\n    \\\"contract_address\\\": \\\"0xbF9f4B85CCeC7d129257868C59d8968F210bCd7e\\\",\\n    \\\"raw_contract_code\\\": \\\"0x73bf9f4b85ccec7d129257868c59d8968f210bcd7e301460806040526004361060335760003560e01c806396e4ee3d146038575b600080fd5b606b60048036036040811015604c57600080fd5b8101908080359060200190929190803590602001909291905050506081565b6040518082815260200191505060405180910390f35b600081830290509291505056fea26469706673582212208e6ae6bd6c413f17a418a0a76d70cae2ef4dadb9e5851bd2aacbae2a2e16226c64736f6c63430006000033\\\",\\n    \\\"state\\\": {},\\n    \\\"static_references\\\": [],\\n    \\\"state_references\\\": []\\n}\"\n    ],\n    \"state_references\": []\n}"
      +    "{\n    \"contract_address\": \"0x663144906fD032B4A3F8c8726c8Eb36EC9c860B5\",\n    \"raw_contract_code\": \"0x608060405234801561001057600080fd5b50600436106100415760003560e01c80637bd703e81461004657806390b98a111461009e578063f8b2cb4f14610104575b600080fd5b6100886004803603602081101561005c57600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff16906020019092919050505061015c565b6040518082815260200191505060405180910390f35b6100ea600480360360408110156100b457600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff16906020019092919080359060200190929190505050610200565b604051808215151515815260200191505060405180910390f35b6101466004803603602081101561011a57600080fd5b81019080803573ffffffffffffffffffffffffffffffffffffffff169060200190929190505050610359565b6040518082815260200191505060405180910390f35b600073bf9f4b85ccec7d129257868c59d8968f210bcd7e6396e4ee3d61018184610359565b60026040518363ffffffff1660e01b8152600401808381526020018281526020019250505060206040518083038186803b1580156101be57600080fd5b505af41580156101d2573d6000803e3d6000fd5b505050506040513d60208110156101e857600080fd5b81019080805190602001909291905050509050919050565b6000816000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff1681526020019081526020016000205410156102515760009050610353565b816000803373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002060008282540392505081905550816000808573ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff168152602001908152602001600020600082825401925050819055508273ffffffffffffffffffffffffffffffffffffffff163373ffffffffffffffffffffffffffffffffffffffff167fddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef846040518082815260200191505060405180910390a3600190505b92915050565b60008060008373ffffffffffffffffffffffffffffffffffffffff1673ffffffffffffffffffffffffffffffffffffffff16815260200190815260200160002054905091905056fea26469706673582212209e9b636358d088d2b7d86e0b4f5736e27f616eb19d64df8a156584ef93d6fb6f64736f6c63430006000033\",\n    \"state\": {\n        \"c02b974894de5bf0fb7566ce8f7b1adad2756368026c4d4ee0415f5e0642cc68\": \"0000000000000000000000000000000000000000000000000000000000002710\"\n    },\n    \"static_references\": [\n        \"{\\n    \\\"contract_address\\\": \\\"0xbF9f4B85CCeC7d129257868C59d8968F210bCd7e\\\",\\n    \\\"raw_contract_code\\\": \\\"0x73bf9f4b85ccec7d129257868c59d8968f210bcd7e301460806040526004361060335760003560e01c806396e4ee3d146038575b600080fd5b606b60048036036040811015604c57600080fd5b8101908080359060200190929190803590602001909291905050506081565b6040518082815260200191505060405180910390f35b600081830290509291505056fea2646970667358221220e9607140e10b82ef7fb798a72a1c186d3b32da0498cba4bd8b75163e547884fe64736f6c63430006000033\\\",\\n    \\\"state\\\": {},\\n    \\\"static_references\\\": [],\\n    \\\"state_references\\\": []\\n}\"\n    ],\n    \"state_references\": []\n}"
         ]
         "static_references": []
       }
```

**Review by informartin — APPROVED** [review]

**Pull request merged**